src/main/java/teammates/ui/webapi/action/ActionFactory.java

```diff
@@ -43,7 +43,10 @@
         map(ResourceURIs.ACCOUNTS_RESET, PUT, ResetAccountAction.class);
         map(ResourceURIs.COURSE, GET, GetCourseAction.class);
         map(ResourceURIs.COURSE, DELETE, DeleteCourseAction.class);
-        map(ResourceURIs.COURSE, PUT, ArchiveCourseAction.class);
+        map(ResourceURIs.COURSE, POST, AddCourseAction.class);
+        map(ResourceURIs.COURSE, PUT, EditCourseAction.class);
+        //Archive temp. changed to COURSES, not sure how to put it.
+        map(ResourceURIs.COURSES, PUT, ArchiveCourseAction.class);
```

Comment:
Since you are doing course archive, just create a new resource URI to `COURSE_ARCHIEVE`